Design/DesignDoc-Least-ConnectionsTCPLB

```diff
@@ -0,0 +1,181 @@
+
+To demonstrate that feature easily I will use a bucket size of `3` and a rate of `1`.
+
+The public functions of the rate limiter are:
+```go
+type Ratelimiter struct {
+	clients map[string]*client
+	burst   int
+	rate    int
+    mu      sync.Mutex
+}
+func NewRateLimiter(burst, rate int) *Ratelimiter {}
+func (r *Ratelimiter) Allow(clientId string) bool {}
+```
+## Server
+The server uses the forwarder library and appends security.
+### Secured communication
+#### Authentication
+The server uses mutual TLS for authentication. Clients will need to provide a valid client certificate during the TLS handshake.
+
+In Mutual TLS, the server requests the client to provide a digital certificate which contains the client's public key and identity information. The server verifies the client's certificate by checking its authenticity and ensuring it is signed by a trusted Certificate Authority (CA) that the server recognizes. This confirms the client's identity.
+The server also provides its own certificate to the client during TLS handshake. The client verifies the server certificate similarly. This establishes mutual authentication between the client and server.
+
+Certificate rotation is supported by generating new certificates periodically and distributing them to clients. The certificates are distributed before they expire to ensure smooth rotation. When a new certificate is received, the client stops using the old certificate and it can be put in a revocation list. The CRL can be checked by clients during certificate validation and both certificates can be used before using that list.
+##### Certificate configuration
+The server and client certificates must use the same Certificate Authority.
+```
+# Generate CA cert
+openssl genrsa -aes256 -out ca/ca.key 4096 
+openssl req -new -x509 -sha256 -days 20 -key ca/ca.key -out ca/ca.crt
+
+# Generate CSR for server
+openssl genrsa -out server/localhost.key 2048
+openssl req -new -key server/localhost.key -sha256 -out server/localhost.csr
+# Validate CSR
+openssl x509 -req -days 365 -sha256 -in server/localhost.csr -CA ca/ca.crt -CAkey ca/ca.key -set_serial 1 -out server/localhost.crt
+
+# Create client CRT
+openssl genrsa -out client/client1.key 2048
+openssl req -new -key client/client1.key -out client/client1.csr
+# Validate CSR
+openssl x509 -req -days 365 -sha256 -in client/client1.csr -CA ca/ca.crt -CAkey ca/ca.key -set_serial 2 -out client/client1.crt
+```
+
+#### TLS versions
+We will support TLS 1.2 and 1.3 for the communication between clients and server. It removes a lot of unsecure or deprecated TLS features and provides better security. I found out that an estimate of 20% of corporate internet traffic is using TLS 1.3 in 2023 and adding TLS 1.2 will cover the most secure enough clients possible. 
```

Comment:
> We will support TLS 1.2 and 1.3 for the communication between clients and server.

nit: Do you need the 1.2 support? Since you're controlling both support for the older protocol makes is less secure.

Reply:
@jakule , I added this to cover more ground kinda corner case possible as I am aimed to have a detailed Design doc